.github/workflows/e2e-tests.yml

```diff
@@ -0,0 +1,109 @@
+name: E2E Tests
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, labeled]
```

Comment:
we shouldn't rerun e2e tests every time the PR is labeled, it'll be pretty costly (e.g. just now I added the `ready-for-review` label and it triggered the tests). this came up recently in our team as well and Tarun mentioned looking into https://github.com/orgs/community/discussions/27031#discussioncomment-3254376